packages/ui/components/data-table/DataTableSelectionBar.tsx

```diff
@@ -21,10 +30,16 @@ export function DataTableSelectionBar<TData>({ table, actions }: DataTableSelect
   return (
     <div className="bg-brand-default text-brand item-center absolute bottom-0 left-1/2 flex -translate-x-1/2 gap-4 rounded-lg p-2">
       <div className="text-brand-subtle my-auto px-2">{numberOfSelectedRows} selected</div>
-      {actions?.map((action) => (
-        <Button aria-label={action.label} onClick={action.onClick} StartIcon={action.icon} key={action.label}>
-          {action.label}
-        </Button>
+      {actions?.map((action, index) => (
+        <Fragment key={index}>
+          {action.type === "action" ? (
+            <Button aria-label={action.label} onClick={action.onClick} StartIcon={action.icon}>
+              {action.label}
+            </Button>
+          ) : action.type === "render" ? (
+            action.render(table)
```

Comment:
Pass table as a callback -> Pass state out of the DataTableComponent and into the user sheet for example